src/dmd/cparse.d

```diff
@@ -2754,7 +2754,7 @@ final class CParser(AST) : Parser!AST
         bool dllimport;  // TODO implement
         bool dllexport;  // TODO implement
 
-        if (token.value == TOK.identifier)
+        if (isGnuAttributeName())
```

Comment:
Right, so have left this in place then, and just added an else condition to consume the unhandled attribute.  It could be better, as `__attribute__(( const() ))` is accepted, but `__attribute__(( const (()) ))` is not - I'd err towards delaying proper argument handling till later though.